go.mod

```diff
@@ -0,0 +1,3 @@
+module github.com/ethereum/evmc/v7
```

Comment:
Why is there a need for v7 here? If we bump to ABI 8 and this is changed to v8, will go modules still be able to download v7 things?

Or the only benefit is knowing it is v7 even if someone uses a commithash after it?

Reply:
This is required by `go mod`. The `vX` is required if you version is `vX.Y.Z`. Can be only omitted if `X==1 || X==0`.

Versions are marked with git tags, and v7 series will continue to work (because e.g. at `v7.3.0` this line will be still `module github.com/ethereum/evmc/v7`.

The usage is something like this:

```
go get github.com/ethereum/evmc/v7@v7.3.0
go get github.com/ethereum/evmc/v7@v7.3.1
go get github.com/ethereum/evmc/v8@v8.0.0
```

And also 
```go
import "github.com/ethereum/evmc/v7"
import "github.com/ethereum/evmc/v8"
```

So users need to update their import directives when upgrading major version number. In case of EVMC where vX matches ABI version this is reasonable IMHO.

Reply:
So it is just a convention required by `go mod`, it won't magically make `v7` work when we move on to `v8`, because it still just downloads from the repo master, right?

Reply:
1. `go get github.com/ethereum/evmc/v7@v8.0.0` will fail due to mismatch.
2. `go get github.com/ethereum/evmc/v7@master` will also fail if the most recent version is `v8.Y.Z`. `go mod` uses a kind of `git describe` strategy to figure out the "current" version.

Reply:
Sounds good.